index.src.html

```diff
@@ -1810,73 +1818,269 @@
 </dl>
 
 
-Raw Media Interfaces (Frames){#raw-media-interfaces}
+Raw Media Interfaces {#raw-media-interfaces}
 ====================================================
 These interfaces represent unencoded (raw) media.
 
+Memory Model {#raw-media-memory-model}
+--------------------------------------
 
-AudioFrame Interface {#audioframe-interface}
+### Background ### {#raw-media-memory-model-background}
+
+This section is non-normative.
+
+Decoded media data may occupy a large amount of system memory. To minimize the
+need for expensive copies, this specification defines a scheme for reference
+counting (`clone()` and `close()`).
+
+### Reference Counting ### {#raw-media-memory-model-reference-counting}
+
+A <dfn>media resource</dfn> is storage for the actual pixel data or the audio
+sample data described by a {{VideoFrame}} or {{AudioData}}.
+
+The {{AudioData}} {{AudioData/[[resource reference]]}} and {{VideoFrame}}
+{{VideoFrame/[[resource reference]]}} internal slots hold a reference to a
+[=media resource=].
+
+{{VideoFrame}}.{{VideoFrame/clone()}} and
+{{AudioData}}.{{AudioData/clone()}} return new frame objects who's
+`[[resource reference]]` points to the same [=media resource=] as the original
+frame.
+
+{{VideoFrame}}.{{VideoFrame/close()}} and {{AudioData}}.{{AudioData/close()}}
+will clear their [[resource reference]] slot, releasing the reference their
+[=media resource=]
+
+A [=media resource=] must remain alive at least as long as it continues to be
+referenced by a `[[resource reference]]`.
+
+NOTE: When a [=media resource=] is no longer referenced by a
+    `[[resource reference]]`, the resource may be destroyed. User agents are
+    encouraged to destroy such resources quickly to reduce memory pressure and
+    facilitate resouce reuse.
+
+AudioData Interface {#audiodata-interface}
 ---------------------------------------------
 
 <xmp class='idl'>
 [Exposed=(Window,DedicatedWorker)]
-interface AudioFrame {
-  constructor(AudioFrameInit init);
+interface AudioData {
+  constructor(AudioDataInit init);
+
+  readonly attribute AudioSampleFormat sampleFormat;
+  readonly attribute float sampleRate;
+  readonly attribute unsigned long numberOfFrames;
+  readonly attribute unsigned long numberOfChannels;
+  readonly attribute unsigned long allocationSize;
+  readonly attribute unsigned long long duration;
   readonly attribute unsigned long long timestamp;
-  readonly attribute AudioBuffer? buffer;
+
+  undefined copyFromChannel(BufferSource destination, unsigned long channelNumber);
```

Comment:
Something that occurred to me right after the call last Friday is that this API is not very good for interleaved data, it's really wasteful: each byte is touched twice to copy the full buffer.

Reply:
Yeah I would expect this to be something like copyFromPlane and interleaved data would only have one plane.